### ISSUE TITLE
Failed monster sanity checks exposed by fuzzer.

### ISSUE THREAD

**Issue by NullCGT** [NullCGT/SpliceHack-Rewrite#44]
This is a very persistent bug, and I have not been able to discover the source. It occurs fairly frequently when fuzzing, and yet I've never found it during regular gameplay.

Sample output:

```

Suddenly, the dungeon collapses.
illegal mon data 555555a03568; mnum=317 (fmon)
  Generating more information you may report:

[00] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(+0xea318) [0x55555563e318]
[01] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(+0xeaad1) [0x55555563ead1]
[02] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(panic+0x26e) [0x555555640ea8]
[03] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(+0x15fb30) [0x5555556b3b30]
[04] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(mon_sanity_check+0x6c) [0x5555556b44b0]
[05] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(sanity_check+0x23) [0x5555555e000a]
[06] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(moveloop+0x11d9) [0x5555555b312e]
[07] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(main+0x8b2) [0x5555557eb29b]
[08] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7ffff7c340b3]
[09] /home/username/Workspace/SpliceHack-Rewrite/src/splicehack(_start+0x2e) [0x5555555b1c8e]

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7c32859 in __GI_abort () at abort.c:79
#2  0x000055555563eac8 in NH_abort () at end.c:206
#3  0x0000555555640ea8 in panic (str=str@entry=0x555555876918 "illegal mon data %s; mnum=%d (%s)") at end.c:705
#4  0x00005555556b3b30 in sanity_check_single_mon (mtmp=mtmp@entry=0x555555b2ab00, chk_geno=chk_geno@entry=1 '\001', msg=msg@entry=0x555555833eb9 "fmon") at mon.c:63
#5  0x00005555556b44b0 in mon_sanity_check () at mon.c:205
#6  0x00005555555e000a in sanity_check () at cmd.c:3012
#7  0x00005555555b312e in moveloop (resuming=<optimized out>) at allmain.c:493
#8  0x00005555557eb29b in main (argc=<optimized out>, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:335
```

Any help with this issue would be greatly appreciated.

**Comment by NullCGT** [comment]
Confirmed with Nero that this is a monster template bug caused by improper mons array accesses.

**Comment by RojjaCebolla** [comment]
Nicely done. And thanks, Nero!

**Comment by NullCGT** [comment]
Sanity check fix here: b3e8aa54a31d87c7a2ac4e35f169d4617e48842a.
